ndpyramid/reproject.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import annotations  # noqa: F401
+
+import typing
+from collections import defaultdict
+
+import datatree as dt
+import numpy as np
+import xarray as xr
+from rasterio.warp import Resampling
+
+from .common import Projection
+from .utils import (
+    add_metadata_and_zarr_encoding,
+    get_levels,
+    get_version,
+    multiscales_template,
+)
+
+
+def _da_reproject(da, *, dim, crs, resampling, transform):
+    if da.encoding.get('_FillValue') is None and np.issubdtype(da.dtype, np.floating):
+        da.encoding['_FillValue'] = np.nan
+    return da.rio.reproject(
+        crs,
+        resampling=resampling,
+        shape=(dim, dim),
+        transform=transform,
+    )
+
+
+def level_reproject(
+    ds: xr.Dataset,
+    *,
+    projection: typing.Literal['web-mercator', 'equidistant-cylindrical'] = 'web-mercator',
```

Comment:
This type hinting is neat! I like that it saves a check on the projection being on of a set of values.